__tests__/app.test.js

```diff
@@ -54,12 +54,319 @@ describe.only("GET /api/articles/:article_id", () => {
+      },
+      {
+        comment_id: 9,
+        votes: 0,
+        created_at: "2020-01-01T03:08:00.000Z",
+        author: "icellusedkars",
+        body: "Superficially charming",
+      },
+      {
+        comment_id: 12,
+        votes: 0,
+        created_at: "2020-03-02T07:10:00.000Z",
+        author: "icellusedkars",
+        body: "Massive intercranial brain haemorrhage",
+      },
+      {
+        comment_id: 13,
+        votes: 0,
+        created_at: "2020-06-15T10:25:00.000Z",
+        author: "icellusedkars",
+        body: "Fruit pastilles",
+      },
+      {
+        comment_id: 18,
+        votes: 16,
+        created_at: "2020-07-21T00:20:00.000Z",
+        author: "butter_bridge",
+        body: "This morning, I showered for nine minutes.",
+      },
+    ];
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((res) => {
+        expect(res.body).toEqual(expectedCommentArray);
+      });
+  });
+  test("404 : /api/9999 -> route that does not exist: 404 Not Found", () => {
+    return request(app)
+      .get("/api/articles/9999/comments")
+      .expect(404)
+      .then(({ body: { message } }) => {
+        expect(message).toBe("not found");
+      });
+  });
+  test("422 : /api/articlezz -> invalid path", () => {
+    return request(app)
+      .get("/api/articles/articlezz/comments")
+      .expect(422)
+      .then(({ body: { message } }) => {
+        expect(message).toBe("Invalid path");
+      });
+  });
+});
+
+describe("POST /api/articles/:article_id/comments", () => {
+  const commentToPost = {
+    username: "rogersop",
+    body: "Lobster Pot",
+  };
+
+  test("415 : Invalid body /missing properties", () => {
+    const invalidDataBody = {
+      1: "rogersop",
+      2: "lobster Pot",
+    };
+    const missingDataBody = {
+      1: "rogersop",
+    };
+
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(invalidDataBody)
+      .expect(415)
+      .then(({ body: { message } }) => {
+        expect(message).toBe("Unsupported Data/Body sent");
+      });
+  });
+
+  test("200 : send request body with username and body", () => {
+    return request(app)
+      .post("/api/articles/4/comments")
+      .send(commentToPost)
+      .expect(200)
+      .then(({ body: { postedComment } }) => {
+        expect(postedComment).toEqual(
+          expect.objectContaining({
+            comment_id: 19,
+            body: "Lobster Pot",
+            article_id: 4,
+            author: "rogersop",
+            votes: 0,
+          })
+        );
+      });
+  });
+
+  test("422 : Invalid article id /api/articles/not-an-id/comments", () => {
+    return request(app)
+      .post("/api/articles/not-an-id/comments")
+      .send(commentToPost)
+      .expect(422)
+      .then(({ body: { message } }) => {
+        expect(message).toBe("Invalid id");
+      });
+  });
+
+  test("400 : bad request /api/articles/99999/comments", () => {
```

Comment:
404 not found is better for a Article ID that is valid but just doesn't exist in our db